// careless-salted/Button.js
var Careless = require('careless');

module.exports = function layout(props, context) {

  var backgroundColor = props.backgroundColor || "#5D9CEC";
  var paddingTop = props.paddingTop || "25px";

  var borderTopBottom = (props.small) ? "10px" : "15px";
  var borderLeftRight = (props.small) ? "20px" : "25px";

  var style = {
    fontSize: (props.small) ? "15px" : "16px",
    fontFamily: "Helvetica, Arial, sans-serif",
    fontWeight: "normal",
    color: "#ffffff",
    textDecoration: "none",
    backgroundColor: backgroundColor,
    borderTop: borderTopBottom + " solid " + backgroundColor,
    borderBottom: borderTopBottom + " solid " + backgroundColor,
    borderLeft: borderLeftRight + " solid " + backgroundColor,
    borderRight: borderLeftRight + " solid " + backgroundColor,
    borderRadius: "3px",
    display: "inline-block"
  };

  return (
    <table width="100%" border="0" cellspacing="0" cellpadding="0" class="mobile-button-container">
      <tr>
        <td align="center" style={"padding: "+paddingTop+" 0 0 0;"} class="padding-copy">
          <table border="0" cellspacing="0" cellpadding="0" class="responsive-table">
            <tr>
              <td align="center">
                <a href={props.href} target="_blank" style={style} class="mobile-button">{props.children}</a>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  )
};
