// careless-salted/Button.js
var Careless = require('careless');

module.exports = function layout(props, context) {

  var backgroundColor = props.backgroundColor || "#5D9CEC";

  var style = {
    fontSize: "16px",
    fontFamily: "Helvetica, Arial, sans-serif",
    fontWeight: "normal",
    color: "#ffffff",
    textDecoration: "none",
    backgroundColor: backgroundColor,
    borderTop: "15px solid " + backgroundColor,
    borderBottom: "15px solid " + backgroundColor,
    borderLeft: "25px solid " + backgroundColor,
    borderRight: "25px solid " + backgroundColor,
    borderRadius: "3px",
    display: "inline-block"
  };

  return (
    <table width="100%" border="0" cellspacing="0" cellpadding="0" class="mobile-button-container">
      <tr>
        <td align="center" style="padding: 25px 0 0 0;" class="padding-copy">
          <table border="0" cellspacing="0" cellpadding="0" class="responsive-table">
            <tr>
              <td align="center">
                <a href={props.href} target="_blank" style={style} class="mobile-button">{props.children}</a>
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>
  )
};
